llama-index-integrations/llms/llama-index-llms-mymagic/llama_index/llms/mymagic/base.py

```diff
@@ -105,11 +106,15 @@ async def _get_result(self, task_id: str) -> Dict[str, Any]:
             return resp.json()
 
     async def acomplete(
-        self, question: str, model: str, max_tokens: int, poll_interval: float = 1.0
+        self,
+        question: str,
+        model: Optional[str] = None,
+        max_tokens: Optional[int] = None,
+        poll_interval: float = 1.0,
     ) -> CompletionResponse:
         self.question_data["question"] = question
-        self.question_data["model"] = model
-        self.question_data["max_tokens"] = max_tokens
+        self.question_data["model"] = model or self.model
+        self.question_data["max_tokens"] = max_tokens or self.max_tokens
```

Comment:
It i required to assign a value to `self.model`:
```suggestion
        self.model = self.question_data["model"] = model or self.model
        self.max_tokens = self.question_data["max_tokens"] = (
            max_tokens or self.max_tokens
        )
```
This is the main readon of the fix, to allow the user to make the model selection.

Reply:
Or we can change `self.model` in `_submit_question` to take `self.question_data["model"]` instead